gt/main.c

```diff
@@ -818,22 +1016,58 @@ gt_proc(void *arg)
 			 */
 			ret = gt_parse_incoming_pkt(m, &pkt_info);
 			if (ret < 0) {
-				switch (pkt_info.outer_ethertype) {
-				case ETHER_TYPE_IPv4:
-					add_pkt_acl(&acl4, m);
-					continue;
-				case ETHER_TYPE_IPv6:
-					add_pkt_acl(&acl6, m);
+				gt_process_unparsed_incoming_pkt(
+					&acl4, &acl6, &num_arp, arp_bufs,
+					m, pkt_info.outer_ethertype);
+				continue;
+			}
+
+			if (cur_tsc - last_tsc >=
+					frag_scan_timeout_ms * tsc_hz) {
```

Comment:
1. Variables `frag_scan_timeout_ms` and `tsc_hz` are only used here. We should combine them to avoid the multiplication for every loop iteration.

2. Let's move this `if` statement to the very end of the loop `while (likely(!exiting)) {`, so we don't interfere with the processing of arriving batches of packets, but in between batches.